apps/test/unit/applab/MockFirebaseTest.js

```diff
@@ -0,0 +1,93 @@
+import { expect } from '../../util/configuredChai';
+import MockFirebase from '../../util/MockFirebase';
+import { getDatabase } from '../../../src/applab/firebaseUtils';
+
+describe('MockFirebase', () => {
+  beforeEach(() => {
+    window.Applab = {
+      channelId: "test-firebase-channel-id",
+      firebaseName: 'test-firebase-name',
+      firebaseAuthToken: 'test-firebase-auth-token',
+    };
+  });
+
+  describe('once', () => {
+    it('calls callbacks', done => {
+      const firebase = new MockFirebase('https://example.firebaseio.com/');
+      firebase.set('foo');
+      firebase.once('value', snapshot => {
+          expect(snapshot.val()).to.equal('foo');
+          done();
+        },
+        error => console.warn(error));
+      firebase.flush();
+    });
+
+    it('resolves promises', done => {
+      const firebase = new MockFirebase('https://example.firebaseio.com/');
+      firebase.set('bar');
+      firebase.once('value').then(snapshot => {
+          expect(snapshot.val()).to.equal('bar');
+          done();
+        },
+        error => console.warn(error));
+      firebase.flush();
+    });
+  });
+
+  describe('set', () => {
+    it('calls callbacks', done => {
+      const firebase = new MockFirebase('https://example.firebaseio.com/');
+      firebase.set('foo', () => done());
```

Comment:
passing `done` directly looks like it will work perfectly in this case, because `done` accepts an error param. however I don't think I can use it in other cases where there are separate success/failure callbacks, because it would presumably interpret the presence of a result passed to the success callback as the presence of an error.